### PR TITLE
[KHRGA-27] Updated CI properties interface

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA-27

### CODE REVIEW

File: source/iface/properties.rst [hunk 1]
```diff
@@ -2,20 +2,192 @@
...
+provide an optional parameter in each of their constructors
+to provide a ``property_list`` which contains zero or more properties.
+Each of those properties augments the semantics of the class with
+a particular feature. Each of those classes must also provide
+``has_property`` and ``get_property`` member functions
+for querying for a particular property.
+
+The example below illustrates how using properties does not affect the
+type of the object, thus, does not prevent the usage of SYCL objects
+in containers.
```
Comment (mkinsner) [inline source/iface/properties.rst:26]:
```suggestion
```

Comment (mkinsner) [reply source/iface/properties.rst:26]: Since the example isn't listed inline, please remove this paragraph

File: source/iface/properties.rst [hunk 2]
```diff
@@ -2,20 +2,192 @@
...
+
+  template <typename Property, typename SyclObject>
+  inline constexpr bool is_property_of_v =
+      is_property_of<Property, SyclObject>::value;
+
+  class T {
+    ...
+
+        template <typename Property>
+        bool has_property() const noexcept;
```
Comment (mkinsner) [inline source/iface/properties.rst:69]:
```suggestion
     template <typename Property>
     bool has_property() const noexcept;
```

Comment (mkinsner) [reply source/iface/properties.rst:69]: Indenting here should match the ... and template below.  I'll update the SYCL spec to correct this

File: source/iface/properties.rst [hunk 3]
```diff
@@ -2,20 +2,192 @@
...
+    ...
+  };
+
+  class property_list {
+   public:
+    template <typename... Properties> property_list(Properties... props);
+  };
+  } // namespace sycl
+
+(constructor)
```
Comment (mkinsner) [inline source/iface/properties.rst:82]:
```suggestion
(constructor) of `sycl::property_list`
```

Comment (mkinsner) [reply source/iface/properties.rst:82]: I don't think it's clear what class the constructor is for.  In the SYCL spec, the table has a caption making the ownership clear

File: source/iface/properties.rst [hunk 4]
```diff
@@ -2,20 +2,192 @@
...
 
 ::
 
-   template <typename... propertyTN>
-   property_list(propertyTN... props);
+  template <typename Property, SyclObject> inline constexpr bool is_property_of_v;
+
+Variable containing value of ``is_property_of<Property, SyclObject>``.
+
+Member functions
```
Comment (mkinsner) [inline source/iface/properties.rst:146]:
```suggestion
Member functions of the SYCL common property interface
```

File: source/iface/properties.rst [hunk 5]
```diff
@@ -2,20 +2,192 @@
...
+  template <typename Property> Property get_property() const
+
+Returns a copy of the property of type ``Property`` that ``T`` was
+constructed with. Must throw an ``exception`` with the
+``errc::invalid`` error code if ``T`` was not constructed
+with the ``Property`` property.
+
+.. _ buffer-properties-example:
+
+``buffer-properties-example``
```
Comment (mkinsner) [inline source/iface/properties.rst:173]:
```suggestion
buffer-properties-example
```

File: source/iface/properties.rst [hunk 6]
```diff
@@ -2,20 +2,192 @@
...
+=============================
```
Comment (mkinsner) [inline source/iface/properties.rst:174]:
```suggestion
=========================
```

File: source/iface/properties.rst (hunk 2 above)
Comment (mkinsner) [reply source/iface/properties.rst:69]: https://github.com/KhronosGroup/SYCL-Docs/pull/493

File: source/iface/properties.rst (hunk 1 above)
Comment (imvucic) [reply source/iface/properties.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

File: source/iface/properties.rst (hunk 2 above)
Comment (imvucic) [reply source/iface/properties.rst]: Followed indenting with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

File: source/iface/properties.rst (hunk 3 above)
Comment (imvucic) [reply source/iface/properties.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

File: source/iface/properties.rst (hunk 4 above)
Comment (imvucic) [reply source/iface/properties.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

File: source/iface/properties.rst (hunk 5 above)
Comment (imvucic) [reply source/iface/properties.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

File: source/iface/properties.rst (hunk 6 above)
Comment (imvucic) [reply source/iface/properties.rst]: Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/90/commits/995fa62042bc32dcef75f3ac099f68fab5c818c1

**Review by mkinsner — APPROVED** [review]

**Pull request merged**